### PR TITLE
fix: added missing prisma imports

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Required to unblock #11789 

Adds missing prisma imports and refactors checks a little bit.

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- yarn build --filter=@calcom/api...

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by zomars** [comment]
Current dependencies on/for this PR:
* main
  * **PR #11867** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/11867" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/11867?utm_source=stack-comment).

**Review by zomars — COMMENTED** [review]
@alishaz-polymath What do you think about this refactor? I'm trying to encapsulate as much in the request as possible.

Also added missing prisma imports that are caught thanks to this PR #11839

File: apps/api/pages/api/event-types/_post.ts [hunk 1]
```diff
@@ -284,8 +284,8 @@ async function postHandler(req: NextApiRequest) {
   await checkPermissions(req);
 
   if (parsedBody.parentId) {
-    await checkParentEventOwnership(parsedBody.parentId, userId);
-    await checkUserMembership(parsedBody.parentId, parsedBody.userId);
+    await checkParentEventOwnership(req);
+    await checkUserMembership(req);
```
Comment (zomars) [inline apps/api/pages/api/event-types/_post.ts:288]: We could also pass parsedBody as down below but I like the self-containment here. https://github.com/calcom/cal.com/pull/11867/files#diff-53284e5334f4b512fe926bb35af4b182fc262e5a2a0dda8ac2f6c1d418163b86R295

**Review by alishaz-polymath — APPROVED** [review]
🚀 
Brilliant improvements 🙏

File: apps/api/pages/api/event-types/_post.ts (hunk 1 above)
Comment (alishaz-polymath) [reply apps/api/pages/api/event-types/_post.ts:288]: Yeah, I agree. This is better 🙏

**Pull request merged**